instrumentation/internal/internal-lambda/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/lambda/LambdaTransformer.java

```diff
@@ -34,29 +24,25 @@ public static byte[] transform(byte[] classBytes, String slashClassName, Class<?
     if (InjectedClassHelper.isHelperClass(targetClass)) {
       return classBytes;
     }
+    ClassFileTransformer transformer = ClassFileTransformerHolder.getLambdaClassFileTransformer();
+    if (transformer == null) {
+      return classBytes;
+    }
 
-    ClassFileTransformer transformer = ClassFileTransformerHolder.getClassFileTransformer();
-    if (transformer != null) {
-      try {
-        byte[] result;
-        if (IS_JAVA_9) {
-          result =
-              Java9LambdaTransformer.transform(
-                  transformer, classBytes, slashClassName, targetClass);
-        } else {
-          result =
-              transformer.transform(
-                  targetClass.getClassLoader(), slashClassName, null, null, classBytes);
-        }
-        if (result != null) {
-          return result;
-        }
-      } catch (Throwable throwable) {
-        // sun.instrument.TransformerManager catches Throwable from ClassFileTransformer and ignores
-        // it, we do the same.
+    try {
+      // The 'targetClass' needs to be non-null as it is used in Java9LambdaClassFileTransformer
+      // to get a reference to the jpms module. However, a null value must be passed to the
+      // underlying transformer as this code is called when the lambda is defined.
+      byte[] result =
+          transformer.transform(
```

Comment:
I didn't immediately get why you wanted to change this, but I guess it is to reduce the helper classes and push more logic into the agent code. I think this is good idea and getting rid for the extra module for java9 is nice. What I'd change is that I'd probably also rename `ClassFileTransformerHolder` as it is only for lambdas as you noted. Also I'd consider not using `ClassFileTransformer` here. Passing the `targetClass` argument to the transform method confused me a bit as `transform` should have the class there only when retransforming/redefining which we aren't doing here. You could have an interface in `ClassFileTransformerHolder` that takes the necessary arguments. What do you think?

Reply:
I agree on the `ClassFileTransformerHolder` rename as it's definitely only used for lambdas.

I initially tried to simplify the design/API but that requires to introduce a new interface, so I thought maybe keep reusing `ClassFileTransformer` could be an acceptable compromise, but I agree that it might be a bit confusing as the implementations details leak into the callers as shown by the quantity of comments to use it the way it's intended.

I'll do another attempt by introducing a new interface to see if it helps improve this.